discussion/filter.go

```diff
@@ -0,0 +1,64 @@
+package discussion
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+	validate.RegisterTagNameFunc(func(fld reflect.StructField) string {
+		name := strings.SplitN(fld.Tag.Get("json"), ",", 2)[0]
+		if name == "-" {
+			return ""
+		}
+		return name
+	})
+}
+
+type Filter struct {
+	Type          string `json:"type" validate:"omitempty,oneof=openended issues qanda all"`
+	State         string
+	Assignees     []string
+	Assets        []string
+	Owner         string
+	Labels        []string
+	SortBy        string `json:"sort" validate:"omitempty,oneof=created_at updated_at"`
+	SortDirection string `json:"direction" validate:"omitempty,oneof=asc desc"`
+	Size          int
+	Offset        int
+}
+
+// Validate will check whether fields in the filter fulfills the constraint
+func (f *Filter) Validate() error {
+
+	if f.Size < 0 {
```

Comment:
is there a reason on why are we not using validate annotation like some of the fields?

Reply:
Just want to make the error specific and not returning error from Validator to the user.